### ISSUE TITLE
pwn.conf skips validation checks

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#1033]
This blows up when the target architecture is 32-bit (e.g. `i386`) since we assemble a 64-bit object file and attempt to link it to a 32-bit ELF.

This is a somewhat expected result, since i386-64 doesn't exist.

```
/usr/local/bin/i386-unknown-linux-gnu-ld: i386:x86-64 architecture of input file `/var/folders/4h/rrwj8fpj1cqcfb83syr_f930008rd2/T/pwn-asm-eSdmd1/step2-obj' is incompatible with i386 output
```

While this is a user error, we should probably warn about this scenario.  It likely applies in lots of situations -- I think the best place to handle this is in the `LocalContext` decorator.  This ensures that we don't explode immediately on `context.bits` assignment (since it is valid to set `bits` before `arch`), but the user gets warned.

**Comment by zachriggle** [comment]
Apparently this is a result of `.pwn.conf` not triggering all of the validation routines, since we just replace the defaults.

**Comment by Arusekk** [comment]
This was fixed in #1034
